src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -21,6 +40,13 @@ public NativeBufferPool(int maxElementsInBuffer)
             _buckets = new NativeBufferBucket<T>[maxBuckets + 1];
         }
 
+        public void Dispose()
+        {
+            foreach (NativeBufferBucket<T> bucket in _buckets)
```

Comment:
what happens if `_buckets` is or becomes `null`? The implied contract is that `IDispose.Dispose()` should never throw.

Try 
```
var local = null;
if ((local = Interlocked.Exchange(ref _buckets, null)) != null)
{
    foreach (NativeBufferBucket<T> bucket in local)
        // same as before
}
```